src/queries/method_parameter_count_changed.ron

```diff
@@ -0,0 +1,68 @@
+SemverQuery(
+    id: "method_parameter_count_changed",
+    human_readable_name: "pub method parameter count changed",
+    description: "Parameter count of a method has changed.",
+    required_update: Major,
+    reference_link: Some("https://doc.rust-lang.org/cargo/reference/semver.html#fn-change-arity"),
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on ImplOwner {
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+                        name @output @tag
+
+                        importable_path {
+                            path @output @tag
+                        }
+
+                        inherent_impl {
+                            method {
+                                method_visibility: visibility_limit @filter(op: "=", value: ["$public"]) @output
+                                method_name: name @output @tag
+
+                                parameters @fold @transform(op: "count") @output @tag(name: "parameters")
+
+                                span_: span @optional {
+                                    filename @output
+                                    begin_line @output
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            current {
+                item {
+                    ... on ImplOwner {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        name @filter(op: "=", value: ["%name"])
+
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                        }
+
+                        # We use "impl" instead of "inherent_impl" here because moving
+                        # an inherently-implemented method to a trait is not necessarily
+                        # a breaking change, so we don't want to report it.
+                        impl {
```

Comment:
For bonus points and if isn't too much work, consider adding a test case for this to guard against a possible future regression. If we accidentally were to change this `impl` to `inherent_impl`, this could result in a false-positive.

False-positives are particularly dangerous to the `cargo-semver-checks` project as a whole, because they dilute the confidence users have in the tool, and may lead to users to stop using it or not adopt it in the first place.

Reply:
Actually, the comment was misleading: I copied it from inherent_method_missing query, but now have I realised that's not the case here. If there is no matching function on the current crate, the error won't get reported. Nevertheless, I will add some more test cases.

Reply:
Don't we still want to catch cases where a method got moved from an inherent impl to a trait impl *and also* had its parameter count change? If this is `inherent_impl` instead of `impl` then that case sounds like it wouldn't get reported.

Reply:
Yes, I was thinking of a scenario, where a method is moved from an inherent impl to a trait impl and the parameter count doesn't change. Then, changing `impl` to `inherent_impl` wouldn't generate an error, as there is no match in the current crate for the baseline version of the method.

Reply:
Right. But leaving `inherent_impl` would also not generate an error because the parameter count wouldn't have changed, right? So it seems like `inherent_impl` is the better option, and the comment is not out of place?